addons/fleet/models/fleet_vehicle_cost.py

```diff
@@ -59,7 +59,7 @@ def compute_next_year_date(self, strdate):
     def _compute_contract_name(self):
         for record in self:
             name = record.vehicle_id.name
```

Comment:
```suggestion
            name = record.vehicle_id.name or ""
```

I think this should be the patch instead. because if name is "", you still would want to add the cost_subtype_id.name and with your patch you can't.

Reply:
@MiquelRForgeFlow https://github.com/odoo/odoo/blob/4e1034ff2f08fa20bcdc9ac39eb2489d34a9715d/addons/fleet/models/fleet_vehicle.py#L98

record.vehicle_id.name will always have a name. (Here vehicle_id is False as user has not yet selected vehicle_id)

> you still would want to add the cost_subtype_id.name 

When user will save the record, It will be computed correctly.

This patch is to avoid traceback in a case when user select contract and Type before selecting Vehicle